modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/ContentDashboardAdminPortletGetPropsTest.java

```diff
@@ -0,0 +1,1138 @@
+				_company.getGroupId(), TestPropsValues.getUserId());
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.addVocabulary(
+				TestPropsValues.getUserId(), _company.getGroupId(),
+				RandomTestUtil.randomString(), serviceContext);
+		AssetVocabulary childAssetVocabulary =
+			_assetVocabularyLocalService.addVocabulary(
+				TestPropsValues.getUserId(), _company.getGroupId(),
+				RandomTestUtil.randomString(), serviceContext);
+
+		try {
+			JournalTestUtil.addArticle(
+				TestPropsValues.getUserId(), _group.getGroupId(), 0);
+
+			Map<String, Object> data = _getData(
+				_getMockLiferayPortletRenderRequest(
+					new String[] {
+						assetVocabulary.getName(),
+						childAssetVocabulary.getName()
+					}));
+
+			Map<String, Object> props = (Map<String, Object>)data.get("props");
+
+			Assert.assertNotNull(props);
+
+			JSONArray vocabulariesJSONArray = (JSONArray)props.get(
+				"vocabularies");
+
+			Assert.assertEquals(
+				String.valueOf(JSONFactoryUtil.createJSONArray()),
+				vocabulariesJSONArray.toString());
+		}
+		finally {
+			_assetVocabularyLocalService.deleteAssetVocabulary(assetVocabulary);
+			_assetVocabularyLocalService.deleteAssetVocabulary(
+				childAssetVocabulary);
+		}
+	}
+
+	@Test
+	public void testGetPropsWithMissingChildAssetVocabulary() throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				_company.getGroupId(), TestPropsValues.getUserId());
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.addVocabulary(
+				TestPropsValues.getUserId(), _company.getGroupId(),
+				"vocabulary", serviceContext);
+
+		AssetCategory assetCategory = _assetCategoryLocalService.addCategory(
+			TestPropsValues.getUserId(), _company.getGroupId(), "category",
+			assetVocabulary.getVocabularyId(), serviceContext);
+
+		AssetVocabulary childAssetVocabulary =
+			_assetVocabularyLocalService.addVocabulary(
+				TestPropsValues.getUserId(), _company.getGroupId(),
+				RandomTestUtil.randomString(), serviceContext);
+
+		try {
+			JournalTestUtil.addArticle(
+				_group.getGroupId(), 0,
+				_getServiceContext(
+					TestPropsValues.getUserId(), _group.getGroupId(),
+					new long[] {assetCategory.getCategoryId()}));
+
+			Map<String, Object> data = _getData(
+				_getMockLiferayPortletRenderRequest(
+					new String[] {
+						assetVocabulary.getName(),
+						childAssetVocabulary.getName()
+					}));
+
+			Map<String, Object> props = (Map<String, Object>)data.get("props");
+
+			Assert.assertNotNull(props);
+
+			JSONArray vocabulariesJSONArray = (JSONArray)props.get(
+				"vocabularies");
+
+			Assert.assertEquals(
+				JSONUtil.put(
+					JSONUtil.put(
+						"key", String.valueOf(assetCategory.getCategoryId())
+					).put(
+						"name", "category"
+					).put(
+						"value", 1L
+					).put(
+						"vocabularyName", "vocabulary"
+					)
+				).toString(),
+				vocabulariesJSONArray.toString());
```

Comment:
Maybe it is clarer if we use String.valueof(props.get("vocabularies"))

Reply:
uhm, yes, good catch, I'll give it a try ;)